components/VercelChat/tools/browser/ExtractResultView.tsx

```diff
@@ -0,0 +1,95 @@
+import { cn } from "@/lib/utils";
+import { formatFieldName, formatFieldValue, isPriorityField } from "@/lib/browser/resultFormatters";
+
+const STYLES = {
+  text: {
+    primary: "text-gray-900 dark:text-gray-100",
+    muted: "text-gray-600 dark:text-gray-400",
+  },
+} as const;
+
+const MAX_DISPLAYED_FIELDS = 8;
+
+// Runtime guard: Ensure value is a plain object
+function isPlainObject(value: unknown): value is Record<string, unknown> {
+  return (
+    typeof value === 'object' && 
+    value !== null && 
+    !Array.isArray(value) && 
+    Object.getPrototypeOf(value) === Object.prototype
+  );
+}
```

Comment:
Single Responsibility Principle
- actual: multiple functions defined in a single file
- required: exactly one function defined per file